lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -185,28 +185,31 @@ def package_manager
       end
 
       def os_facts
-        facter = ForemanMaintain::Utils::Facter.path
-        @os_facts ||= JSON.parse(execute("#{facter} -j os"))
+        @os_facts ||= ForemanMaintain::Utils::OsFacts
       end
 
       def el?
-        os_facts['os']['family'] == 'RedHat'
+        ids = ['rhel', 'rhel fedora']
+        ids.include?(os_facts.id_like) ||
+          ids.include?(os_facts.id)
       end
 
       def debian?
-        os_facts['os']['family'] == 'Debian'
+        ids = %w[debian ubuntu]
+        ids.include?(os_facts.id_like) ||
+          ids.include?(os_facts.id)
```

Comment:
It's probably find to use this (provided `id_like` becomes an array)
```suggestion
        os_facts.id == 'debian'  || os_facts.id_like.include?('debian')
```

It may also be sufficient to check if `/etc/debian_version` exists.

Reply:
+1 on `/etc/debian_version`

Reply:
In general I dont think `/etc/debian_version` can be missing on system. However I felt to keep `/etc/os-release` if in user error case that is missing.